src/DynamoCore/Graph/Nodes/VariableInputNode.cs

```diff
@@ -173,14 +174,16 @@ private void MarkNodeDirty()
         /// </summary>
         public virtual void RemoveInputFromModel()
         {
-            var count = model.InPorts.Count;
-            if (count > 0)
+            int count = model.InPorts.Count;
+            // Do not remove input port if there aren't any
+            // or if the node is a DSVarArgFunction and doing so would remove a default port
+            bool countIsGreaterThanDefault = !(model is DSVarArgFunction dSVarArgFunction) || count > dSVarArgFunction.DefaultNumInputs;
+            if (count != 0 && countIsGreaterThanDefault)
             {
                 var port = model.InPorts[count - 1];
                 port.DestroyConnectors();
                 model.InPorts.Remove(port);
             }
-
             MarkNodeDirty();
```

Comment:
@StudioLE @mjkkirschner One thing I found from my local testing of this is that, even with the default state and user would not delete default ports, but node still marked dirty which I think is not necessary because that will trigger a graph run. Can we make another PR and put the statement in the `if` above?